klippy/extras/manual_stepper.py

```diff
@@ -65,11 +74,23 @@ def do_move(self, movepos, speed, accel, sync=True):
                           accel_t, cruise_t, accel_t,
                           cp, 0., 0., axis_r, 0., 0.,
                           0., cruise_v, accel)
-        self.next_cmd_time = self.next_cmd_time + accel_t + cruise_t + accel_t
-        self.rail.generate_steps(self.next_cmd_time)
-        self.trapq_finalize_moves(self.trapq, self.next_cmd_time + 99999.9)
-        toolhead = self.printer.lookup_object('toolhead')
-        toolhead.note_kinematic_activity(self.next_cmd_time)
+        next_print_time = self.next_cmd_time + accel_t + cruise_t + accel_t
+        while 1:
+            segment_time = self.next_cmd_time + batch_time
+            self.next_cmd_time = min(segment_time, next_print_time)
+            self.rail.generate_steps(self.next_cmd_time)
+            finalize_time = self.next_cmd_time - FLUSH_DELAY
+            if self.next_cmd_time >= next_print_time:
+                finalize_time = self.reactor.NEVER
+            self.trapq_finalize_moves(self.trapq, finalize_time)
+            toolhead.note_kinematic_activity(self.next_cmd_time)
+            for m in self.all_mcus:
+                flush_time = self.next_cmd_time
+                if self.next_cmd_time < next_print_time:
+                    flush_time -= self.move_flush_time
```

Comment:
It might make sense to move flush_time calculation outside of 'for m in ...' loop, as it does not depend on the mcu. So it might be a bit easier to understand in the future.

Reply:
But I'm afraid that now the code is incorrect. The whole block of `flush_time` calculation should have been moved outside the loop, otherwise the code
```
                if self.next_cmd_time < next_print_time:
                    flush_time -= self.move_flush_time
```
will be executed for each mcu, and so flush_time will be decreasing for each mcu.